app/controllers/api/v1/spots_controller.rb

```diff
@@ -3,6 +3,7 @@ module V1
     class SpotsController < ApiController
       def create
         @spot = current_user.spots.create!(spot_params)
+        save_compatible_matches
```

Comment:
Try to keep your controller as thin as possible. See the [guide](https://github.com/rootstrap/tech-guides/blob/master/ruby/rails.md#skinny-controllers).

All the business logic must be placed on the model. Remember in Rails we use MVC pattern. So move al this logic to the `Spot` model

Reply:
https://www.sitepoint.com/the-basics-of-mvc-in-rails-skinny-everything/